### PR TITLE
Added Pylint and docformatter

### PR DESCRIPTION
**Pull request by ravi-mosaicml**
Pylint checks for errors not found by other linters, including missing docstrings, public functions that should be private, instance methods that should be class methods or static methods, bad control flow, and other coding issues. 

This PR adds in the pylint config and addresses easy-to-fix errors.

A future PR will enable a pylint check in jenkins. Pylint will be required to pass on all new and modified files in a PR (but not on any existing / grandfathered files).

### CODE REVIEW

**Review by jbloxham — APPROVED** [review]
I don't have any experience with Pylint or docformatter so I'm just trusting that this is a reasonable starting point for the configuration. I reserve judgment for when you actually update the repo to conform to Pylint.

**Comment by hanlint** [comment]
I personally find pylint too strict for a codebase with our purpose, so would prefer `flake8`.

**Comment by Averylamp** [comment]
Personally I believe `pylint` is really well made and much more configurable.  It'll take some time to get the right rules, but I think `pylint` is a great choice and quite well supported

**Comment by ravi-mosaicml** [comment]
The current pylint config is mainly warning on things that need to be addressed -- missing docstrings, unused imports, redefined variables, string lines that were too long, redundant control flow.

We will probably want to tweak the variable name rules some more (e.g. variables like `W`, `H` are fairly common in our codebase) and go with a convention on log formatting.

Here is the current pylint summary on this branch showing which rules were violated (and how many times):
```
Raw metrics
-----------

+----------+-------+------+---------+-----------+
|type      |number |%     |previous |difference |
+==========+=======+======+=========+===========+
|code      |12477  |54.70 |NC       |NC         |
+----------+-------+------+---------+-----------+
|docstring |6001   |26.31 |NC       |NC         |
+----------+-------+------+---------+-----------+
|comment   |779    |3.42  |NC       |NC         |
+----------+-------+------+---------+-----------+
|empty     |3553   |15.58 |NC       |NC         |
+----------+-------+------+---------+-----------+



Duplication
-----------

+-------------------------+------+---------+-----------+
|                         |now   |previous |difference |
+=========================+======+=========+===========+
|nb duplicated lines      |0     |0        |0          |
+-------------------------+------+---------+-----------+
|percent duplicated lines |0.000 |0.000    |=          |
+-------------------------+------+---------+-----------+



Messages by category
--------------------

+-----------+-------+---------+-----------+
|type       |number |previous |difference |
+===========+=======+=========+===========+
|convention |378    |378      |378        |
+-----------+-------+---------+-----------+
|refactor   |134    |134      |134        |
+-----------+-------+---------+-----------+
|warning    |252    |252      |252        |
+-----------+-------+---------+-----------+
|error      |10     |10       |10         |
+-----------+-------+---------+-----------+



Messages
--------

+---------------------------------+------------+
|message id                       |occurrences |
+=================================+============+
|missing-function-docstring       |140         |
+---------------------------------+------------+
|line-too-long                    |77          |
+---------------------------------+------------+
|invalid-name                     |68          |
+---------------------------------+------------+
|logging-fstring-interpolation    |60          |
+---------------------------------+------------+
|missing-class-docstring          |50          |
+---------------------------------+------------+
|no-self-use                      |43          |
+---------------------------------+------------+
|unnecessary-pass                 |42          |
+---------------------------------+------------+
|unused-import                    |30          |
+---------------------------------+------------+
|redefined-builtin                |27          |
+---------------------------------+------------+
|no-else-return                   |21          |
+---------------------------------+------------+
|super-with-arguments             |18          |
+---------------------------------+------------+
|unused-argument                  |13          |
+---------------------------------+------------+
|unspecified-encoding             |12          |
+---------------------------------+------------+
|consider-using-from-import       |11          |
+---------------------------------+------------+
|trailing-whitespace              |10          |
+---------------------------------+------------+
|protected-access                 |10          |
+---------------------------------+------------+
|superfluous-parens               |9           |
+---------------------------------+------------+
|no-self-argument                 |8           |
+---------------------------------+------------+
|consider-iterating-dictionary    |8           |
+---------------------------------+------------+
|useless-object-inheritance       |7           |
+---------------------------------+------------+
|redefined-outer-name             |7           |
+---------------------------------+------------+
|consider-using-with              |7           |
+---------------------------------+------------+
|consider-merging-isinstance      |7           |
+---------------------------------+------------+
|attribute-defined-outside-init   |7           |
+---------------------------------+------------+
|super-init-not-called            |6           |
+---------------------------------+------------+
|f-string-without-interpolation   |6           |
+---------------------------------+------------+
|deprecated-method                |6           |
+---------------------------------+------------+
|dangerous-default-value          |6           |
+---------------------------------+------------+
|consider-using-f-string          |5           |
+---------------------------------+------------+
|useless-super-delegation         |3           |
+---------------------------------+------------+
|use-dict-literal                 |3           |
+---------------------------------+------------+
|raise-missing-from               |3           |
+---------------------------------+------------+
|no-else-raise                    |3           |
+---------------------------------+------------+
|no-else-continue                 |3           |
+---------------------------------+------------+
|consider-using-in                |3           |
+---------------------------------+------------+
|wrong-import-order               |2           |
+---------------------------------+------------+
|use-implicit-booleaness-not-len  |2           |
+---------------------------------+------------+
|unneeded-not                     |2           |
+---------------------------------+------------+
|reimported                       |2           |
+---------------------------------+------------+
|pointless-string-statement       |2           |
+---------------------------------+------------+
|no-else-break                    |2           |
+---------------------------------+------------+
|missing-final-newline            |2           |
+---------------------------------+------------+
|inconsistent-return-statements   |2           |
+---------------------------------+------------+
|import-error                     |2           |
+---------------------------------+------------+
|cell-var-from-loop               |2           |
+---------------------------------+------------+
|broad-except                     |2           |
+---------------------------------+------------+
|unused-variable                  |1           |
+---------------------------------+------------+
|unnecessary-comprehension        |1           |
+---------------------------------+------------+
|ungrouped-imports                |1           |
+---------------------------------+------------+
|unbalanced-tuple-unpacking       |1           |
+---------------------------------+------------+
|trailing-comma-tuple             |1           |
+---------------------------------+------------+
|too-many-lines                   |1           |
+---------------------------------+------------+
|lost-exception                   |1           |
+---------------------------------+------------+
|expression-not-assigned          |1           |
+---------------------------------+------------+
|consider-using-set-comprehension |1           |
+---------------------------------+------------+
|consider-using-dict-items        |1           |
+---------------------------------+------------+
|chained-comparison               |1           |
+---------------------------------+------------+
|arguments-renamed                |1           |
+---------------------------------+------------+
|abstract-method                  |1           |
+---------------------------------+------------+
```

Going to merge this in for now as it addresses a lot of extra whitespace, but going to leave pylint enforcement disabled.

File: composer/algorithms/augmix/augmix.py [hunk 1]
```diff
@@ -39,8 +39,9 @@ def augment_and_mix(img: Optional[ImageType] = None,
                     width: int = 3,
                     alpha: float = 1.0,
                     augmentation_set: List = augmentation_sets["all"]) -> ImageType:
-    """Applies AugMix (`Hendrycks et al. <http://arxiv.org/abs/1912.02781>`_) data augmentation to an image.
-    See :class:`AugMix` for details.
+    """Applies AugMix (`Hendrycks et al.
+
+    <http://arxiv.org/abs/1912.02781>`_) data augmentation to an image. See :class:`AugMix` for details.
```
Comment (jbloxham) [inline composer/algorithms/augmix/augmix.py:44]: 👀

**Pull request merged**